src/daemon/daemon.cpp

```diff
@@ -307,6 +309,49 @@ auto grpc_status_for(fmt::memory_buffer& errors)
                         fmt::format("The following errors occurred:\n{}", error_string), "");
 }
 
+auto get_unique_id(const mp::Path& data_path)
+{
+    QFile id_file{QDir(data_path).filePath(uuid_file_name)};
+    QString id;
+
+    if (!id_file.exists())
+    {
+        id = mp::MetricsProvider::generate_unique_id();
+        id_file.open(QIODevice::WriteOnly);
+        id_file.write(id.toUtf8());
+    }
+    else
+    {
+        id_file.open(QIODevice::ReadOnly);
+        id = QString(id_file.readAll());
```

Comment:
Not sure if we should be worried about it since it's just an anonymous randomly generated identifier, but I suppose if the value read from the file is not a valid UUID, then we should generate a new one and save and use that.

Reply:
I suppose that would be true for any of the values we store... Let's get to that when we build the config system.

Reply:
Ack